### PR TITLE
Implement storage host functions

### PR DESCRIPTION
**Pull request by stoyan-lime**
Implemented storage host functions
Some non-static methods similar to static ones have been added. Static ones will be removed when other HostAPIs have been implemented.
To be added in separate PR:
- Fixing reading and writing from memory
- DB transactions

### CODE REVIEW

File: src/main/java/com/limechain/runtime/hostapi/HostApi.java [hunk 1]
```diff
@@ -94,4 +94,25 @@ protected static ByteBuffer getByteBuffer(Memory memory) {
...
+        ByteBuffer buffer = getByteBuffer(memory);
+        int position = buffer.position();
+        buffer.put(position, data, 0, data.length);
+        buffer.position(position + data.length);
+
+        return new RuntimePointerSize(position, data.length);
+    }
+
+    public void putDataToMemoryBuffer(RuntimePointerSize runtimePointerSize, byte[] data) {
+        //TODO: implement
```
Comment (georg-getz) [inline src/main/java/com/limechain/runtime/hostapi/HostApi.java:116]: This is not in the scope of this PR?

Comment (stoyan-lime) [reply src/main/java/com/limechain/runtime/hostapi/HostApi.java:116]: Yes, a separate PR will cover memory operations.

File: src/main/java/com/limechain/runtime/hostapi/RuntimePointerSize.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    public RuntimePointerSize(Number pointerSize) {
+        this(pointerSize.longValue());
+    }
+    public int pointer() {
+        return (int) this.pointerSize;
+    }
+
+    public int size() {
+        return (int) (pointerSize >> 32);
+    }
```
Comment (georg-getz) [inline src/main/java/com/limechain/runtime/hostapi/RuntimePointerSize.java]: 32 is a magic number. 
Always add more inline documentation when working with bitwise operators as the code is harder to read, `pointer & 0xffffffffL` should be explained.

File: src/main/java/com/limechain/storage/KVRepository.java [hunk 3]
```diff
@@ -35,13 +37,36 @@ public interface KVRepository<K, V> {
...
-     * Deletes a key-value pair from the DB for the n(limit) key starting with prefix
+     * Deletes key-value pairs from the DB where key starts with prefix, up to a given limit.
      *
-     * @param prefixSeek prefix for the key of the pair
+     * @param prefix prefix for the key of the pair
      * @param limit maximum entries to delete
-     * @return whether the delete operation was successful
+     * @return how many entries were deleted and if all were deleted
+     */
+    DeleteByPrefixResult deleteByPrefix(String prefix,@Nullable Long limit);
```
Comment (georg-getz) [inline src/main/java/com/limechain/storage/KVRepository.java]: nit: could add soace after the comma

**Review by georg-getz — APPROVED** [review]

**Pull request merged**